File: 7-kyu/Two Oldest Ages/twoOldestAges.go
// Kata can be found at: https://www.codewars.com/kata/511f11d355fe575d2c000001
//Primera vez que usé Golang, sin tener mucha idea de él.

package kata

func TwoOldestAges(ages []int) [2]int {

   var arr [2]int;
   temp := 0;
   
   for i := 1; i < len(ages); i++  {     
     
     if (ages[temp] < ages[i])  {
     
        temp = i;
     
     }
     
   }
   
   arr[0] = ages[temp];
   ages[temp] = -1;
   
   temp = 0;
   
   for i := 0; i < len(ages); i++   {
     
     if (ages[temp] < ages[i])  {
     
        temp = i;
     
     }
   
   }
   
  arr[1] = ages[temp];
  temp = arr[0];
  arr[0] = arr[1];
  arr[1] = temp;
  
  return arr;

}

File: 7-kyu/Two Oldest Ages/twoOldestAges.js
// Kata can be found at: https://www.codewars.com/kata/511f11d355fe575d2c000001


function twoOldestAges(ages){

  var a = 0;
  var b = 0;
  
  for(var i = 0; i < ages.length; i++)  {
  
    if(ages[i] > b)  {
    
      a = b; 
      b = ages[i];
    
    }  else if(ages[i] > a)  {
    
      a = ages[i];
    
    }
  
  }
  var arr = [2];
  arr[0] = a;
  arr[1] = b;
  return arr;

}
